kauri/kauri.go

```diff
@@ -0,0 +1,320 @@
+// Package kauri contains the implementation of the kauri protocol
+package kauri
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"math/rand"
+	"reflect"
+	"sort"
+	"time"
+
+	"github.com/relab/gorums"
+	"github.com/relab/hotstuff"
+	"github.com/relab/hotstuff/backend"
+	"github.com/relab/hotstuff/eventloop"
+	"github.com/relab/hotstuff/internal/proto/hotstuffpb"
+	"github.com/relab/hotstuff/internal/proto/kauripb"
+	"github.com/relab/hotstuff/logging"
+	"github.com/relab/hotstuff/modules"
+)
+
+func init() {
+	modules.RegisterModule("kauri", New)
+}
+
+// TreeType determines the type of tree to be built, for future extension
+type TreeType int
+
+// NoFaultTree is a tree without any faulty replicas in the tree, which is currently supported.
```

Comment:
These comments should be moved inside the const block above each of the constants so that the docs are generated correctly.